pkg/controller/perconaservermongodb/mgo.go

```diff
@@ -238,6 +256,47 @@ const (
 	`
 )
 
+func (r *ReconcilePerconaServerMongoDB) handleRsAddToShard(m *api.PerconaServerMongoDB, replset *api.ReplsetSpec, rspod corev1.Pod,
+	mongosPods []corev1.Pod) error {
+	if len(mongosPods) != int(m.Spec.Sharding.Mongos.Size) {
+		return errors.New("not all mongos pods run")
+	}
+
+	for _, pod := range mongosPods {
+		if !isContainerAndPodRunning(rspod, "mongod") || !isPodReady(rspod) {
+			return errors.New("rsPod is not redy")
```

Comment:
Should we check all containers before the loop? Otherwise, some nodes could be added to the shard while some not.

Reply:
Then it will be added to shard on next reconcile loop